LibraryApp/src/components/books-list.js

```diff
@@ -1,41 +1,133 @@
 import {FilterListBySearch} from 'helpers/library';
-import {search, qrCode} from 'constants/icons';
+import {search, qrCode, refresh} from 'constants/icons';
 import List from 'components/shared/list';
+import {LocationStatus} from 'constants/location-status';
 import {useHardwareBack} from 'components/custom/hardware-back';
 import {CustomImage} from 'components/shared/common';
 
 const isIOSPlatform = Platform.OS == 'ios';
 const loaderTimeout = 2000;
+const defaultObj = {};
 
 const {
   flex1,
   rowFlex,
+  spaceBetween,
   borderRadius20,
   horizontalCenter,
   verticalCenter,
 } = Global;
-const {comicFont, gochiFont, iconFont, fs18, fs20, fs25, bold} = Typography;
+const {
+  comicFont,
+  gochiFont,
+  iconFont,
+  fs16,
+  fs18,
+  fs20,
+  fs25,
+  bold,
+} = Typography;
 const {p10, py20, px10, px15, px25, p15, mtAuto, mb15} = Spacing;
-const {blue, darkGrey, purple, lightGreen, maroon, white} = Colors;
+const {blue, darkGrey, purple, lightGreen, maroon, skyBlue, white} = Colors;
+const {loading, loaded, denied} = LocationStatus;
+
+const Options = {
+  currentLocation: {
+    enableHighAccuracy: false,
+    timeout: 30000,
+    maximumAge: 1000,
+  },
+  subscribeLocation: {
+    enableHighAccuracy: false,
+    maximumAge: 1000,
+    distanceFilter: 1,
+  },
+};
 
 const getItemKey = (item) => `book-${item.id}`;
 
 const BooksList = (props) => {
   const [isRefreshing, setIsRefreshing] = useState(false);
   const [searchedText, setSearchedText] = useState('');
+  const [locationStatus, setLocationStatus] = useState('');
+  let watchID;
+
+  useEffect(() => {
+    requestLocationPermission();
+    return () => {
+      if (watchID) {
+        Geolocation.clearWatch(watchID);
+      }
+    };
+  }, []);
+
+  const requestLocationPermission = () => {
+    if (isIOSPlatform) {
+      getOneTimeLocation();
+      subscribeLocation();
+    } else {
+      PermissionService.requestAndroidLocation().then((result) => {
+        if (result == 'granted') {
+          getOneTimeLocation();
+          subscribeLocation();
+        } else {
+          setLocationStatus(denied);
+        }
+      });
+    }
+  };
+
+  const getOneTimeLocation = () => {
+    const {setCurrentLocation} = props;
+    setLocationStatus(loading);
+    Geolocation.getCurrentPosition(
+      // ------ Will give you the current location ------ //
+      (position) => {
+        setLocationStatus(loaded);
+        const currentLongitude = JSON.stringify(position.coords.longitude);
+        const currentLatitude = JSON.stringify(position.coords.latitude);
+        setCurrentLocation(currentLatitude, currentLongitude);
+      },
+      (error) => {
+        setLocationStatus(error.message);
+      },
+      Options.currentLocation,
+    );
+  };
+
+  // ---------- Will give you the location on location change --------- //
+  const subscribeLocation = () => {
+    const {setCurrentLocation} = props;
+    watchID = Geolocation.watchPosition(
+      (position) => {
```

Comment:
see if you can take out common code in a separate method

Reply:
Done